gobblin-hive-registration/src/main/java/gobblin/hive/HiveRegister.java

```diff
@@ -212,7 +212,7 @@ public abstract boolean existsPartition(String dbName, String tableName, List<Co
    * @param tableName the table name
    * @throws IOException
    */
-  public abstract void dropTableIfExists(String dbName, String tableName) throws IOException;
+  public abstract void dropTableIfExists(String dbName, String tableName, String azkabanUrl, String metastoreURI) throws IOException;
```

Comment:
You shouldn't modify public APIs. Furthermore, these parameters shouldn't be part of this method. The HiveRegister implementation must already know the metastoreURI, while the azkabanUri should be injected into the EventSubmitted.